### ISSUE TITLE
Add-ons already belonging to the collection should not be listed among search suggestions

### ISSUE THREAD

**Issue by AlexandraMoga** [mozilla/addons-frontend#5126]
STR:

1. Log in to AMO-dev
2. Go to the details page of one of your existing collections and click on Edit
3. In the edit collection screen, search for an add-on to add to your collection (i.e. 'tree')
4. Search for the same add-on again and look at the search results

Actual result:
The add-on that was just added to the collection is still listed among search results

Expected result:
An add-on that is already part of the edited collection should not be listed among search results 

Notes:
- currently, an error message is displayed when the same add-on is added to the collection but I think that eliminating the add-on from search results altogether would make more sense
- reproduced on AMO-dev with FF60, Win10x64

![collection search](https://user-images.githubusercontent.com/31961530/40642449-cdf4c79e-6325-11e8-8fb1-a9b4871768d3.gif)

**Comment by bobsilverberg** [comment]
The current Autocomplete API, which we use for this feature, has no way of filtering out results from a given collection. We could tackle this either by enhancing that API to allow a collection id to be passed in, which it would use to exclude add-ons from that collection, or we could handle it on the front-end by further filtering the results after they are returned by the API, but I think that is less desirable.

@eviljeff What do you think would be a good approach to this problem?

**Comment by eviljeff** [comment]
use `exclude_addons=...` - pretty much all the filters available in the main search API endpoint are available.  http://addons-server.readthedocs.io/en/latest/topics/api/addons.html#search

**Comment by bobsilverberg** [comment]
I did some investigation into this, and we will have a problem using `exclude_addons` because we only ever have a single page of add-ons for a collection at any given point in time. For this reason I have opened https://github.com/mozilla/addons-server/issues/8925 which requests a way to specify a collection whose add-ons should be excluded from search results.

**Comment by bobsilverberg** [comment]
As suggested at https://github.com/mozilla/addons-server/issues/8925#issuecomment-406092937, a possible approach would be to retrieve 50 add-ons when a user is editing a collection, and use that list of 50 add-ons to exclude them from the search suggestions. For a collection with more than 50 add-ons, we might still end up with add-ons in the search suggestions which already exist in the collection, but for any collection with 50 or less add-ons, this issue would be fully addressed. Just waiting on feedback from @jvillalobos and @pwalm (requested at https://github.com/mozilla/addons-server/issues/8925#issuecomment-406092937) before proceeding with this implementation.

**Comment by bobsilverberg** [comment]
Based on the discussion at https://github.com/mozilla/addons-server/issues/8925 I'm going to close this as wontfix. The message displayed to a user is straightforward and unambiguous, and we do not deem fixing this worth the time and effort it would take.
